lib/crystalball/predictor/modified_support_specs.rb

```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+
+require 'crystalball/predictor/helpers/affected_examples_detector'
+
+module Crystalball
+  class Predictor
+    # Used with `predictor.use Crystalball::Predictor::ModifiedSupportSpecs.new`. Will find files that match passed regexp and
+    # return full spec files which uses matched support spec files. Perfectly works for shared_context and shared_examples.
+    class ModifiedSupportSpecs
+      include ::Crystalball::Predictor::Helpers::AffectedExamplesDetector
+
+      # @param [Regexp] support_spec_pattern - regexp to filter support specs files
+      def initialize(support_spec_pattern = %r{spec/support/.*\.rb\z})
```

Comment:
Should not this default be somewhere in settings?

Reply:
It is a common way to configure predictors right now. I think it makes sense to move it to config file later